### DISCUSSION TITLE
Is there a way to write relations directly in Go using the DB transaction I already have open.

### DISCUSSION THREAD

**Discussion by advdv** [authzed/spicedb#1840] [Q&A] [answered]
I know this is probably just something that we can do because we're a Go shop but it would be nice if we can write the relations to the SpiceDB tables inside the same transactions that we already use to store the rest of our data. This would give us "perfect" consistency between the relations data and the rest of our data.

Is there an abstraction in the SpiceDB codebase that can be used for this? I understand it would be more low level, but maybe there are some pointers/links to the codebase I can check out for this?

**Comment by josephschorr** [comment, marked as answer]
@advanderveer Not currently, no. There is a readwrite transaction abstraction, but it explicitly does not expose access to the underlying SQL, but rather to an interface for only making relationship changes

**Reply by advdv** [reply]
Understandable,  thank you for the quick response

**Reply by josephschorr** [reply]
If you have a proposal, we'd be quite happy to see it
